### PR TITLE
Improve hexbin() documentation

### PR DESCRIPTION
**Pull request by timhoffm**
- See-also-link between hexbin() and hist2d()
- More compact example code notation
- Better example Introduction

### CODE REVIEW

File: examples/statistics/hexbin_demo.py [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-Hexbin Demo
-===========
+=====================
+Hexagonal binned plot
+=====================
 
-Plotting hexbins with Matplotlib.
+`~.Axes.hexbin` is a 2D histogram plot, in which the bins are hexagons.
+This is a kind of density plot, representing the number of data points within
+a bin by color.
```
Comment (jklymak) [inline examples/statistics/hexbin_demo.py:8]:
```suggestion
`~.Axes.hexbin` is a 2D histogram plot in which the bins are hexagons and 
the color represents the number of data points within each bin.
```

File: examples/statistics/hexbin_demo.py [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
 
-Hexbin is an axes method or pyplot function that is essentially
-a pcolor of a 2D histogram with hexagonal cells.  It can be
-much more informative than a scatter plot. In the first plot
-below, try substituting 'scatter' for 'hexbin'.
+Hexagonal bins have less sampling bias from edge effects than rectangular
+bins and can thus be a more correct representation of the data.
```
Comment (jklymak) [inline examples/statistics/hexbin_demo.py:11]:
```suggestion
For some data distributions, hexagonal bins may have less sampling bias from edge effects than rectangular bins.
```

... though I'd just not say this at all, unless you have a citation to say that hexagonal bins are "more correct"

**Review by jklymak — COMMENTED** [review]
Looks good, but a couple of small comments!

File: examples/statistics/hexbin_demo.py (hunk 2 above)
Comment (timhoffm) [reply examples/statistics/hexbin_demo.py:10]: > Hexagons are the maximum number of sides a polygon can havefor a regular tesselation of the plane, so in terms of packing a hexagon is 13%more efficient for covering the plane than squares.  This property translates intobetter sampling efficiency at least for elliptical shapes.  Lastly hexagons are vi-sually less biased for displaying densities than other regular tesselations.  Forinstance with squares our eyes are drawn to the horizontal and vertical lines ofthe grid. 

https://cran.r-project.org/web/packages/hexbin/vignettes/hexagon_binning.pdf

Comment (jklymak) [reply examples/statistics/hexbin_demo.py:10]: I agree for the example shown there, where the size of the bin is proportional to the density in the bin.  Less convinced that it is true for a normal hex bin plot.

Comment (timhoffm) [reply examples/statistics/hexbin_demo.py]: Removed.

**Review by jklymak — APPROVED** [review]

**Pull request merged**